api/v1alpha3/machinedeployment_webhook_test.go

```diff
@@ -30,6 +30,13 @@ func TestMachineDeploymentDefault(t *testing.T) {
 		ObjectMeta: metav1.ObjectMeta{
 			Name: "test-md",
 		},
+		Spec: MachineDeploymentSpec{
+			Selector: metav1.LabelSelector{
+				MatchLabels: map[string]string{
+					ClusterLabelName: "test-cluster",
+				},
+			},
```

Comment:
Specifying the Selector here is causing the test failure: line 49 is checking for the MachineDeploymentLabelName key which is only set if there is not a Selector already defined.

It might be good to add a separate test that specifies md.Spec.Selector.MatchLabels and md.Spec.Template.Labels and verifies that the MachineDeployment labels are not added, while the ClusterLabelName ones are.